app/src/org/commcare/views/notifications/NotificationMessage.java

```diff
@@ -83,6 +104,20 @@ public boolean equals(Object o) {
             return false;
         }
 
+        if((nm.buttonInfo == null) != (this.buttonInfo == null)) {
+            //One has buttonInfo while the other doesn't
+            return false;
+        }
+        if(nm.buttonInfo != null) {
+            if(!nm.buttonInfo.getButtonText().equals(this.buttonInfo.getButtonText())) {
+                return false;
+            }
+
+            if(!nm.buttonInfo.getButtonAction().equals(this.buttonInfo.getButtonAction())) {
+                return false;
+            }
+        }

```

Comment:
move this inside a `equals` method in button info class and here simply compare the 2 button info objects.

Reply:
Agreed